doc/debugger.rst

```diff
@@ -922,6 +935,8 @@ Error codes
 +--------+------------------------------------------------------------------+
 | 0x03   | Not found (e.g. invalid breakpoint index)                        |
 +--------+------------------------------------------------------------------+
+| 0x04   | AppRequest error (i.e. request callback returned -1)             |
```

Comment:
Unknown error is used for just various argument errors and such now, for example:

    duk_debug_write_error_eom(thr, DUK_DBG_ERR_UNKNOWN, "invalid heapptr");

It wouldn't hurt to have a fine grained error code, but I'm even more hesitant about adding a rich integer error code space than I'm about adding integer commands :) I'd actually rather extend the error syntax to have:

- A very sparse error code
- A programmatic string error code (for example, `"NOT_FOUND"`) which would be easy to extend without conflict
- A freeform string

Anyway, adding a new error code seems unnecessary unless it requires specific handling in the debug client.